pandas/io/parsers/c_parser_wrapper.py

```diff
@@ -79,6 +82,10 @@ def __init__(self, src: ReadCsvBuffer[str], **kwds) -> None:
             kwds.pop(key, None)
 
         kwds["dtype"] = ensure_dtype_objs(kwds.get("dtype", None))
+        dtype_backend = get_option("mode.dtype_backend")
+        kwds["dtype_backend"] = dtype_backend
+        if dtype_backend == "pyarrow":
+            import_optional_dependency("pyarrow")  # Ensure that we fail here
```

Comment:
"Ensure that we fail here" means "in the case where pyarrow is not installed, better to fail loudly here than somewhere else"?

Reply:
Yep, will clarify the comment